p2p/client_test.go

```diff
@@ -63,34 +87,204 @@ func TestBootstrapping(t *testing.T) {
 	require.NotEmpty(cid2)
 
 	// client1 has no seeds
-	client1, err := NewClient(config.P2PConfig{ListenAddress: "/ip4/127.0.0.1/tcp/7676"}, privKey1, logger)
+	client1, err := NewClient(context.Background(), config.P2PConfig{ListenAddress: "/ip4/0.0.0.0/tcp/7676"}, privKey1, logger)
```

Comment:
Consider using [mocked network](https://github.com/libp2p/go-libp2p/tree/master/p2p/net/mock). It provides a full in-mem implementation for Host networking without OS sockets interactions. Also, it drastically simplifies the testing codebase, as it handles keys/ids/addressees. However, that would require you not to create a Host within the Client itself, but to receive it in the constructor, which is generally a good idea for testing and avoiding tight coupling.

Reply:
Created separate issue for this: https://github.com/lazyledger/optimint/issues/26